### DISCUSSION TITLE
Cannot use SLEAP shortcut keys when using XFCE terminal

### DISCUSSION THREAD

**Discussion by carson2stoker** [talmolab/sleap#831] [Help!]
# Bug description
When in the SLEAP GUI, I am trying to add instances and label. Using Alt + clicking on a node and scrolling does not do anything. I did it before on windows, but now I am using it on Linux remotely on a high performance cluster (because I have access to a GPU). 

## Expected behaviour
I thought the instance would rotate.

## Actual behaviour
The instance does not move at all. I can see no other changes

## Your personal set up
I am working remotely on nodes from the Center for High Performance Computing at the University of Utah. I use FastX3 to access this and am opening an XFCE terminal. The node has a GPU on it. 

*Actually I tried also opening a XFCE desktop. When I did that, it still would not let me rotate an instance, but additionally, I could not even move a whole instance using ALT + dragging a node. When I attempted to do that it just exited sleap gui out of fullscreen and dragged the gui screen around. 

- OS: Linux-4.18.0-348.20.1.el8_5.x86_64-x86_64-with-centos-8.5-Green_Obsidian

- Version(s):
-  SLEAP: 1.2.4
TensorFlow: 2.6.3
Numpy: 1.19.5
Python: 3.7.12

- SLEAP installation method (listed [here](https://sleap.ai/installation.html#)):
  - [ ] [Conda from package](https://sleap.ai/installation.html#conda-package)

<details><summary>Environment packages</summary>

```
# paste output of `pip freeze` or `conda list` here
```

</details>

<details><summary>Logs</summary>

```
(base) [u1208563@notchpeak1:~]$ salloc --time=3:00:00 --account=notchpeak-gpu --partition=notchpeak-gpu --nodes=1 --ntasks=1 --mem=3000  --gres=gpu:1
salloc: Pending job allocation 4893493
salloc: job 4893493 queued and waiting for resources
salloc: job 4893493 has been allocated resources
salloc: Granted job allocation 4893493
salloc: Waiting for resource configuration
salloc: Nodes notch271 are ready for job
(base) [u1208563@notch271:~]$ module load miniconda3/latest
(base) [u1208563@notch271:~]$ conda activate sleap
(sleap) [u1208563@notch271:~]$ sleap-label
Saving config: /uufs/chpc.utah.edu/common/home/u1208563/.sleap/1.2.4/preferences.yaml
QStandardPaths: error creating runtime directory /run/user/1292742 (Permission denied)
Restoring GUI state...

Software versions:
SLEAP: 1.2.4
TensorFlow: 2.6.3
Numpy: 1.19.5
Python: 3.7.12
OS: Linux-4.18.0-348.20.1.el8_5.x86_64-x86_64-with-centos-8.5-Green_Obsidian

Happy SLEAPing! :)
[aac @ 0x557490c3efc0] Input buffer exhausted before END element found
qt.qpa.xcb: QXcbConnection: XCB error: 3 (BadWindow), sequence: 1998, resource id: 6291462, major code: 40 (TranslateCoords), minor code: 0


# paste relevant logs here, if any
```

</details>

## How to reproduce

1. Open Fastx# for desktop. Create a connection to notchpeak1.chpc.utah.edu.
2. click add session and then XCFE terminal.
3. Use salloc command to move to a compute node on the notchpeak cluster with a GPU.
4. type 'module load miniconda3/latest'
5. type 'conda activate sleap' and then 'sleap-label'
6. open up the existing project
7. add a new instance and attempt to rotate it.

**Comment by roomrys** [comment]
Hi @carson2stoker,

The instances rotate as expected on (local, non-remote) Linux. Thus, the issue likely stems from the `Alt` hotkey for SLEAP being overridden from either the [XFCE hotkeys](https://docs.xfce.org/xfce/xfwm4/getting-started) or the [FastX3 hotkeys](https://www.starnet.com/xwin32kb/keyboard-shortcuts-for-live-sessions/) - both of which use `Alt` in sequence with other keys to call a variety of commands. It seems that XFCE is the issue in this case, per the [XFCE docs](https://docs.xfce.org/xfce/xfwm4/getting-started#move_windows):
> There are several ways to move windows :
-left or right click on the title bar of the window and drag it
-use Alt-left click while the pointer is anywhere over the window frame and move the mouse

I believe you can [set the default shortcut keys used by XFCE](https://docs.xfce.org/faq#is_it_possible_to_change_the_default_shortcut_keys) to a non-SLEAP shortcut key. It could also be that key-presses are not being passed through to SLEAP using FastX3 + XFCE which would require changes to their software or using different software to connect remotely.

Let us know how changing the XFCE shortcut key works - fingers crossed this fixes it. Otherwise, we can help you figure it out.

Thanks,
Liezl
